src/cli/platform/sgx/tcb.rs

```diff
@@ -0,0 +1,199 @@
+/// SGX API documentation: https://api.portal.trustedservices.intel.com/documentation
+/// SGX cert extension documentation: https://download.01.org/intel-sgx/dcap-1.1/linux/docs/Intel_SGX_PCK_Certificate_CRL_Spec-1.1.pdf
+/// Save the resulting files in the `/var/cache/intel-sgx/` directory
+#[derive(Args, Debug)]
+pub struct PckCache {}
+
+impl PckCache {
+    pub fn execute(self) -> anyhow::Result<ExitCode> {
+        if !Path::new(PCKID_CSV_PATH).exists() {
+            Command::new("PCKIDRetrievalTool")
+                .arg("-f")
+                .arg(PCKID_CSV_PATH)
+                .spawn()
+                .expect("Could not run PCKIDRetrievalTool, is it in the $PATH? Are you root?")
+                .wait()?;
+        }
+
+        let report = std::fs::read_to_string(PCKID_CSV_PATH)?;
+        let report_parts: Vec<&str> = report.split(',').collect();
+        let url = format!(
+            "{FMSPC_URL}/?encrypted_ppid={}&cpusvn={}&pcesvn={}&pceid={}",
+            report_parts[0], report_parts[2], report_parts[3], report_parts[1]
+        );
+
+        let (pck_cert, _) = fetch_file(&url).context("Failed to fetch PCK certificate")?;
+        let mut pck_cursor = std::io::Cursor::new(pck_cert);
+
+        let pck_cert = rustls_pemfile::certs(&mut pck_cursor)
+            .context("Failed to PEM-decode PCK certificate")?;
+
+        let pck_cert =
+            Certificate::from_der(&pck_cert[0]).context("Failed to DER-decode PCK certificate")?;
+
+        let fmspc = decode_extension(&pck_cert)?;
+        OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(FMSPC_PATH)
+            .context(format!(
+                "opening destination file {FMSPC_PATH} for saving Intel FMSPC"
+            ))?
+            .write_all(fmspc.as_bytes())
+            .context(format!("writing Intel FMSPC to file {FMSPC_PATH}"))?;
+
+        Ok(ExitCode::SUCCESS)
+    }
+}
+
+fn decode_extension(cert: &Certificate) -> anyhow::Result<String> {
+    const SGX_EXT_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.113741.1.13.1");
+    const FMSPC_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.113741.1.13.1.4");
+    const FMSPC_OID_LEN: usize = 10;
+
+    if let Some(extensions) = &cert.tbs_certificate.extensions {
+        for extension in extensions.iter() {
+            if extension.extn_id == SGX_EXT_OID {
+                let offset = extension
+                    .extn_value
+                    .windows(FMSPC_OID_LEN)
+                    .position(|window| window == FMSPC_OID.as_bytes());
+                if let Some(offset) = offset {
+                    // Index into the extension bytes at the offset + OID size + 2
+                    // The extra +2 gets us past the ASN.1 header for these bytes.
+                    let fmspc = &extension.extn_value
+                        [offset + FMSPC_OID_LEN + 2..offset + FMSPC_OID_LEN + 6 + 2];
```

Comment:
Would it make sense to make `6` a const also?
Also make a comment, that the ASN.1 header size is known, because the inner size is known.

Reply:
Ultimately, I would like this SGX Extension handling to be cleaned up better and should probably be in the SGX crate. Issue: https://github.com/enarx/sgx/issues/113.

Reply:
I'd like to defer this until https://github.com/enarx/sgx/pull/115